### PR TITLE
Ensure all images are closed in FlutterImageView

### PR DESCRIPTION
**Pull request by blasten**
## Description

The initial implementation used two references, but this is causing images to remain open. This patch will use a `Queue` to store the images. I added tests to verify that this cannot happen.

## Related Issues

https://github.com/flutter/flutter/issues/63897

## Tests

I added the following tests: Unit tests

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/android/FlutterImageView.java [hunk 1]
```diff
@@ -218,16 +218,15 @@ public void resizeIfNeeded(int width, int height) {
   @Override
   protected void onDraw(Canvas canvas) {
     super.onDraw(canvas);
-    if (nextImage != null) {
+
+    if (!imageQueue.isEmpty()) {
       if (currentImage != null) {
         currentImage.close();
-        pendingImages--;
+        currentImage = null;
```
[the remark this answers is not in the document]
Comment (blasten) [reply shell/platform/android/io/flutter/embedding/android/FlutterImageView.java]: Done.

File: shell/platform/android/io/flutter/embedding/android/FlutterImageView.java [hunk 2]
```diff
@@ -200,15 +211,12 @@ public void resizeIfNeeded(int width, int height) {
     if (width == imageReader.getWidth() && height == imageReader.getHeight()) {
       return;
     }
-    // Close resources.
-    if (nextImage != null) {
-      nextImage.close();
-      nextImage = null;
-    }
+    imageQueue.clear();
     if (currentImage != null) {
```
Comment (jason-simmons) [inline shell/platform/android/io/flutter/embedding/android/FlutterImageView.java:215]: This `if` can be removed now that `currentImage.close()` has been removed

File: shell/platform/android/io/flutter/embedding/android/FlutterImageView.java [hunk 3]
```diff
@@ -70,17 +70,23 @@
...
-    this.kind = kind;
-    init();
+    this(context, createImageReader(width, height), kind);
+  }
+
+  public FlutterImageView(@NonNull Context context) {
+    this(context, 1, 1, SurfaceKind.background);
+  }
+
+  public FlutterImageView(@NonNull Context context, @NonNull AttributeSet attrs) {
```
Comment (jason-simmons) [inline shell/platform/android/io/flutter/embedding/android/FlutterImageView.java:80]: Add the import for `AttributeSet`

Comment (blasten) [reply shell/platform/android/io/flutter/embedding/android/FlutterImageView.java:81]: 🤦‍♂️

File: shell/platform/android/io/flutter/embedding/android/FlutterImageView.java (hunk 2 above)
Comment (blasten) [reply shell/platform/android/io/flutter/embedding/android/FlutterImageView.java]: Done.

**Pull request merged**